### ISSUE TITLE
Fix version.cmake for master branch

### ISSUE THREAD

**Issue by dewyatt** [rnpgp/rnp#1037]
## Description

Our initial release tag (0.9.0) is on a commit shared in the master branch, so rnp master ends up showing a version like:
`0.9.0+git20200114.618.8017915`
Which, although technically accurate, was intended to be more like:
`0.0.0+git20200114.8017915`

**Comment by ni4** [comment]
@dewyatt What about showing it as 0.13.0+git2020114..., i.e. based on the latest release version?

**Comment by dewyatt** [comment]
@ni4 That's how it would show on a release branch but for master that could be considered slightly inaccurate because our current release workflow is kind of based on the possibility that we may have multiple independent release branches (like 0.x series and 1.x series etc) which may have different fixes and features. But that may be overly flexible for us and showing a version based off of the most recent release is more user-friendly, so let me see what I can do.

**Comment by dewyatt** [comment]
I'm going to reopen this as I was never really satisfied with my fix here and it's just causing issues.
What I would propose is:
1. Delete the `v0.9.0` tag. This tag is problematic because this commit is shared between master and the release branch. Alternatively, we could move this tag ahead one commit to 6b0a314 since that's just a doc change. Or we could rename it something like `deprecated-v0.9.0` so it doesn't match the version tag regex.
2. Revert cmake-versioning to the version in 5fd956f and drop the unreliable git branch checking. This means `master` would again report a version of `0.0.0`. Although it's less user friendly, it's probably necessary (see bottom text).
3. Pin rnp to a specific commit of cmake-versioning from here on.

Basically we maintain two versioning schemes. For release branches we have a version like 0.13.0. For branches that do not have tags (`master` *should* be included here), we use the commit timestamp. You can see how this is utilized in ruby-rnp for checking various quirks [here](https://github.com/rnpgp/ruby-rnp/blob/361c6b8f18437ea29bab666d5479a80bfb8ec924/lib/rnp/misc.rb#L273). My current kludge using `VERSION_OVERRIDE_BRANCH` is faulty on PRs because of how `GITHUB_REF` works there and this causes an incorrect version in the build and thus causes ruby-rnp failures in certain scenarios.

**Comment by dewyatt** [comment]
Since CI is broken, for now I'm planning to proceed with this unless someone objects, in which case I can easily revert the changes.

**Comment by ronaldtse** [comment]
@dewyatt I have no specific opinion here, whatever works is good. Thanks!

**Comment by dewyatt** [comment]
Ok this should be good as of 740093a.
